src/parameters/arrayparam.js

```diff
@@ -0,0 +1,54 @@
+import Param from "./param";
+import { isArray, isEmpty, isFunction, isString } from "../util";
+
+class ArrayParam extends Param {
+  /**
+   * A parameter that represents an array.
+   * @param {string} name - The name of the parameter in snake_case.
+   * @param {string} shortName - The name of the serialized form of the parameter
+   *                         If a value is not provided, the parameter will not be serialized.
+   * @param {string} [sep='.'] - The separator to use when joining the array elements together
+   * @param {function} [process=Util.identity ] - Manipulate origValue when value is called
+   * @class ArrayParam
+   * @extends Param
+   * @ignore
+   */
+  constructor(name, shortName, sep = '.', process) {
+    super(name, shortName, process);
+    this.sep = sep;
+  }
+
+  serialize() {
+    if (this.shortName != null) {
+      let arrayValue = this.value();
+      if (isEmpty(arrayValue)) {
+        return '';
+      } else if (isString(arrayValue)) {
+        return `${this.shortName}_${arrayValue}`;
+      } else {
+        let flat = arrayValue.map(t => isFunction(t.serialize) ? t.serialize() : t).join(this.sep);
+        return `${this.shortName}_${flat}`;
+      }
+    } else {
+      return '';
+    }
+  }
+
+  value() {
+    if (isArray(this.origValue)) {
+      return this.origValue.map(v => this.process(v));
+    } else {
+      return this.process(this.origValue);
```

Comment:
maybe return `this.process(this.origValue) || ''` here?

Reply:
- process() should never return a falsey value here.
- returning an empty string from `value()` is meaningless (it would result in a transformation parameter that doesn't have a value).

If anything goes wrong here it is better to fail fast.